lib/project/setup-project.js

```diff
@@ -11,7 +11,20 @@ const npmConfig = require('../npm/config');
 module.exports = async function setupProject(projectRoot) {
   const config = await npmConfig(projectRoot); // kinda slow, TODO: re-implement as standalone lib
   // const { policies } = options;
-  const pkg = JSON.parse(fs.readFileSync(`${projectRoot}/package.json`, 'utf-8'));
+  const pkgPath = `${projectRoot}/package.json`;
+  if (!fs.existsSync(pkgPath)) {
+    throw new Error(`${pkgPath} does not exist, are you sure this is a valid package?`);
+  }
+  if (!fs.statSync(pkgPath).isFile()) {
```

Comment:
What are are guarding against here specifically? A FIFO queue or device such as `/dev/null` or `/dev/random` or ...? Is this a likely scenario, or truly exceptional

Reply:
That the path is actually a folder. I have a test for this exact scenario in this commit. If you think it's too exceptional, happy to drop.